### PR TITLE
refactor(Vector Tiles): set style in vector tiles parser.

### PR DESCRIPTION
**Pull request by gchoqueux**
## Description

Move vector style parsing in `VectorTileParser`.

## Motivation
* Remove `onParsedFile` in Itowns.
* do all the geometry and style job in  `VectorStyleParse` and avoid a next step to add style.
* Remove Loop on all features to add Style.

### CODE REVIEW

File: src/Provider/DataSourceProvider.js [hunk 1]
```diff
@@ -26,6 +26,7 @@ export function parseSourceData(data, extDest, layer) {
         withNormal: layer.isGeometryLayer !== undefined,
         withAltitude: layer.isGeometryLayer !== undefined,
         layers: source.layers,
+        styles: source.styles,
```
Comment (gmaillet) [inline src/Provider/DataSourceProvider.js:29]: update the documentation for this styles option

Comment (gchoqueux) [reply src/Provider/DataSourceProvider.js:29]: doc added

**Review by gmaillet — APPROVED** [review]

File: src/Provider/DataSourceProvider.js [hunk 2]
```diff
@@ -26,6 +26,7 @@ export function parseSourceData(data, extDest, layer) {
...
         style: layer.style,
```
Comment (zarov) [inline src/Provider/DataSourceProvider.js:30]: I am not a fan of having`styles` and `style` on the same object. Any idea on an alternative here ?

Comment (gchoqueux) [reply src/Provider/DataSourceProvider.js:30]: yes, it will move in future PR.

**Pull request merged**